### PR TITLE
bug: add sphinx extension substitutions

### LINKED ISSUE #28: Install substitutions extension to Sphinx
Sphinx can't substitute variables' values inside directives like `code-block`. Instead, we have to use `parsed-literal` that doesn't support syntax highlight.

There is an extension that enabled the missing functionality: https://github.com/adamtheturtle/sphinx-substitution-extensions
The installation instruction in the repo doesn't match what we have in our project, so need to figure our how to install it.

Additional work needs to be done to replace all current `parsed-literal` instances with `code-block`

### PR DESCRIPTION
**Pull request by rollandf**
Fixes #28

### CODE REVIEW

**Comment by rollandf** [comment]
@noama-nv PTAL

File: docs/getting-started-kubernetes.rst [hunk 1]
```diff
@@ -1948,6 +1963,7 @@ Prerequisites for Worker Nodes
 Supported operating systems:
 
 * Ubuntu 22.04
+* RHEL 8.8
```
Comment (adrianchiris) [inline docs/getting-started-kubernetes.rst:1966]: auto generated ?

File: Makefile [hunk 2]
```diff
@@ -112,9 +113,24 @@ api-docs: gen-crd-api-docs
 	docker run --rm --volume "`pwd`:/data:Z" pandoc/minimal -f html -t rst --lua-filter=/data/hack/ref_links.lua \
 	--columns 200 /data/build/_output/crds-api.html -o /data/docs/customizations/crds.rst
 
+.PHONY: build-cache
+build-cache:
+	@if [ -d "$(CACHE_DIR)" ]; then \
+		echo "$(CACHE_DIR) exists. Skipping build-cache."; \
```
Comment (adrianchiris) [inline Makefile:119]: can you add a clean target which will delete this cache dir ?

**Review by adrianchiris — APPROVED** [review]
added some minor comments.

overall lgtm.

i dont know much about packman, so i trust this works. you can self merge once addressed

File: Makefile (hunk 2 above)
Comment (rollandf) [reply Makefile:119]: Done

File: Makefile [hunk 3]
```diff
@@ -128,3 +144,10 @@ doca-driver-build-update:
 	curl -sL ${DOCA_DRIVER_BUILD_BASE_URL}dtk_nic_driver_build.sh -o $(CURDIR)/docs/downloads/dtk_nic_driver_build.sh
 	curl -sL ${DOCA_DRIVER_BUILD_BASE_URL}RHEL_Dockerfile -o $(CURDIR)/docs/downloads/RHEL_Dockerfile
 	curl -sL ${DOCA_DRIVER_BUILD_BASE_URL}Ubuntu_Dockerfile -o $(CURDIR)/docs/downloads/Ubuntu_Dockerfile
+
+.PHONY: clean
+clean:
+	@echo "Cleaning..."; \
+	rm -rf .cache;\
+	rm -rf build;\
+	rm -rf _build
```
Comment (adrianchiris) [inline Makefile:153]: nit: newline :)

Comment (rollandf) [reply Makefile]: Done

**Pull request merged**